tinyrpc/protocols/jsonrpc.py

```diff
@@ -629,6 +631,12 @@ def parse_reply(
             response.result = rep.get('result', None)
 
         response.unique_id = rep['id']
+        if response.unique_id not in self._incomplete_requests:
+            raise InvalidReplyError(
+                'Reply id does not correspond to any sent requests.'
+            )
```

Comment:
I don't like the reuse of `InvalidReplyError` since it will make it more difficult to distinguish between error causes.
Perhaps subclass `InvalidReplyError` as `UnexpectedIDError` and use this?
That would make it easier to match it in a `raise` statement.

Reply:
That's a good idea. I'll add that in.